src/request/sign-transaction/SignTransactionApi.js

```diff
@@ -4,26 +4,56 @@
 /* global Nimiq */
 /* global AmountTooSmallError, NetworkMissmatchError */
 /* global SignTransactionWithPassphrase, SignTransactionWithPin */
+/* global TransactionType */
+/* global AddressUtils */
 
 class SignTransactionApi extends PopupApi { // eslint-disable-line no-unused-vars
     /**
      * @param {TransactionRequest} txRequest
      */
     async onRequest(txRequest) {
+        // Validation
+        if (!txRequest) throw new Error('Invalid request');
+
+        if (txRequest.type === TransactionType.EXTENDED) {
+            throw new Error('Extended transaction signing is not yet implemented');
+        } else if (txRequest.type === TransactionType.BASIC) {
+            if (txRequest.sender !== txRequest.signer) {
+                throw new Error('Sender must be signer for basic transactions');
+            }
+        } else {
+            throw new Error('Invalid transaction type');
+        }
+
+        // These checks throw an exception on failure
+        AddressUtils.isUserFriendlyAddress(txRequest.sender);
+        AddressUtils.isUserFriendlyAddress(txRequest.recipient);
+        AddressUtils.isUserFriendlyAddress(txRequest.signer);
+
         if (Nimiq.Policy.coinsToSatoshis(txRequest.value) < 1) {
             throw new AmountTooSmallError();
         }
 
         if (txRequest.network !== Nimiq.GenesisConfig.NETWORK_NAME) {
             throw new NetworkMissmatchError(txRequest.network, Nimiq.GenesisConfig.NETWORK_NAME);
         }
+        // End validation
 
+        // Normalization
         txRequest.value = Nimiq.Policy.coinsToSatoshis(txRequest.value);
```

Comment:
can we know for sure that the request value is not in satoshis already?

Reply:
No, but it's defined in the API to be in NIM. Maybe better API design would be to expect in sat, that way an error would mean sending less, not sending more?

Reply:
yep, agree.

Reply:
Sticking to satoshis is generally a good idea.
I was also confused by value and fee being in satoshis in the `txRequest` in `_doSignTransaction` but in NIM here